app/controllers/application_controller.rb

```diff
@@ -18,6 +18,9 @@ class ApplicationController < ActionController::Base
 
   before_action :current_event
   before_action :configure_permitted_parameters, if: :devise_controller?
+  before_action :tracks
```

Comment:
All three of these should not be run for all controllers.  They need to stick to just program controllers for now.